com.unity.multiplayer.mlapi/Tests/Runtime/NetworkPrefabHandlerTests.cs

```diff
@@ -0,0 +1,166 @@
+            networkPrefabHandler.RemoveHandler(baseObject);                     //Remove our handler
+
+            //Register via GlobalObjectIdHash
+            gameObjectRegistered = networkPrefabHandler.AddHandler(baseObject.GlobalObjectIdHash, networkPrefaInstanceHandler);
+
+            //Test result of registering via GlobalObjectIdHash reference
+            Assert.True(gameObjectRegistered);
+
+            //Change it up
+            prefabPosition = new Vector3(6.0f, 4.0f,1.0f);
+            prefabRotation = new Quaternion(3f, 2f, 4f, 1f);
+
+            spawnedObject = networkPrefabHandler.HandleNetworkPrefabSpawn(baseObject.GlobalObjectIdHash, 0, prefabPosition, prefabRotation);
+
+            //Test that something was instantiated
+            Assert.NotNull(spawnedObject);
+
+            //Test that this is indeed an instance of our original object
+            Assert.True(spawnedObject.name.Contains("NetworkPrefabHandlerTestObject"));
+
+            //Test for position and rotation
+            Assert.True(prefabPosition == spawnedObject.transform.position);
+            Assert.True(prefabRotation == spawnedObject.transform.rotation);
+
+            networkPrefabHandler.HandleNetworkPrefabDestroy(spawnedObject);     //Destroy our prefab instance
+            networkPrefabHandler.RemoveHandler(baseObject);                     //Remove our handler
+
+            Assert.False(networkPrefaInstanceHandler.StillHasInstances());
+        }
+
+        [SetUp]
+        public void Setup()
+        {
+            //Create, instantiate, and host
+            NetworkManagerHelper.StartNetworkManager();
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            //Stop, shutdown, and destroy
+            NetworkManagerHelper.ShutdownNetworkManager();
+        }
+    }
+
+
+    /// <summary>
+    /// The Prefab instance handler to use for this test
+    /// </summary>
+    public class NetworkPrefaInstanceHandler : INetworkPrefabInstanceHandler
+    {
+        private NetworkObject m_NetworkObject;
+
+        private List<NetworkObject> m_Instances;
+
+        public NetworkObject HandleNetworkPrefabSpawn(ulong ownerClientId, Vector3 position, Quaternion rotation)
+        {
+            var networkObjectInstance = UnityEngine.Object.Instantiate(m_NetworkObject.gameObject).GetComponent<NetworkObject>();
+            networkObjectInstance.transform.position = position;
+            networkObjectInstance.transform.rotation = rotation;
+            m_Instances.Add(networkObjectInstance);
+            return networkObjectInstance;
+        }
+
+        public void HandleNetworkPrefabDestroy(NetworkObject networkObject)
+        {
+            var instancesContainsNetworkObject = m_Instances.Contains(networkObject);
+            Assert.True(instancesContainsNetworkObject);
+
+            if(instancesContainsNetworkObject)
```

Comment:
no need for this if block at all. it's always going to be true because we're asserting just a line above.